exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java

```diff
@@ -0,0 +1,157 @@
+            //.setMessage(SPAN_DATA_STATUS.getDescription())
+            .build();
+    private static final Span.Links LINKS = Span.Links.newBuilder().setDroppedLinksCount(0).build();
+
+
+    // Some test cases, for example. Since the core function of this exporter is export(), we may focus on testing that.
+
+    @Test
+    public void export(){
+        // Constructs an exporter instance.
+        exporter = new TraceExporter(PROJECT_ID, mockTraceServiceClient, FIXED_ATTRIBUTES);
+        Collection<SpanData> spanDataList = new ArrayList<>();
+
+        //SpanData spanDataOne = SpanData.newBuilder().build();
+
+        SpanData spanDataOne = SpanData.newBuilder()
+                .setParentSpanId(PARENT_SPAN_ID)
+                .setSpanId(SPAN_ID)
+                .setTraceId(TRACE_ID)
+                .setName(SPAN_NAME)
+                .setKind(io.opentelemetry.trace.Span.Kind.SERVER)
+                .setTimedEvents(Collections.emptyList())
+                .setStatus(SPAN_DATA_STATUS)
+                .setStartEpochNanos(START_EPOCH_NANOS)
+                .setEndEpochNanos(END_EPOCH_NANOS)
+                .setTotalRecordedLinks(0)
+                .setHasRemoteParent(false)
+                .setHasEnded(true)
+                .build();
+
+        spanDataList.add(spanDataOne);
+
+        // Some hardcoding of expected data for span.
+        List<Span> expectedSpans = new ArrayList<>();
+        Span spanOne = Span.newBuilder()
+                .setName("projects/" + PROJECT_ID + "/traces/" + TRACE_ID.toLowerBase16() + "/spans/" + SPAN_ID.toLowerBase16())
+                .setSpanId(SPAN_ID.toLowerBase16())
+                .setDisplayName(TruncatableString.newBuilder().setValue(SERVER_PREFIX + SPAN_NAME).setTruncatedByteCount(0).build())
+                .setStartTime(com.google.protobuf.Timestamp.newBuilder().setSeconds(START_SECONDS).setNanos(START_NANOS).build())
+                .setAttributes(ATTRIBUTES)
+                .setTimeEvents(TIME_EVENTS)
+                .setStatus(SPAN_STATUS)
+                .setEndTime(com.google.protobuf.Timestamp.newBuilder().setSeconds(END_SECONDS).setNanos(END_NANOS).build())
+                .setLinks(LINKS)
+                .setParentSpanId(PARENT_SPAN_ID.toLowerBase16())
+                .setSameProcessAsParentSpan(BoolValue.of(true))
+                .build();
+        //Instead of hardcoding as above, Span.parseFrom method can potentially be used to parse from a hardcoded proto
+        expectedSpans.add(spanOne);
+
+        //my original try, but batchWriteSpans is void and cannot be stubbed with a return value
+        //doReturn(expectedSpans).when(mockTraceServiceClient).batchWriteSpans(any(ProjectName.class), anyList());
+
+        //Instead of doReturn, doAnswer is better to deal with whenever batchWriteSpans() is called.
+        //Rather than actually call the BatchWriteSpans() method from the API, null is returned.
+        doAnswer(invocation -> {
```

Comment:
delete and replace with invocation -> null
Like the {} and the comments there

Reply:
Yup, that saves a couple lines